qiskit/circuit/quantumcircuit.py

```diff
@@ -1261,16 +1323,79 @@ def layout(self) -> Optional[TranspileLayout]:
         selected physical qubits on the :class:`~.Target`, and a final layout
         which is an output permutation caused by :class:`~.SwapGate`\s
         inserted during routing.
+
+        Example:
+
+            .. plot::
+                :include-source:
+                :nofigs:
+
+                from qiskit import QuantumCircuit
+                from qiskit.providers.fake_provider import GenericBackendV2
+                from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+
+                # Create circuit to test transpiler on
+                qc = QuantumCircuit(3)
+                qc.h([0, 1, 2])
+
+                # Add measurements to the circuit
+                qc.measure_all()
+
+                # Specify the QPU to target
+                backend = GenericBackendV2(3)
+
+                # Transpile the circuit
+                pass_manager = generate_preset_pass_manager(
+                    optimization_level=1, backend=backend
+                )
+                transpiled_circ = pass_manager.run(qc)
+
+                print(transpiled_circ.layout)
+
+            .. code-block:: text
+
+                TranspileLayout(initial_layout=Layout({
+                0: Qubit(QuantumRegister(3, 'q'), 0),
+                1: Qubit(QuantumRegister(3, 'q'), 1),
+                2: Qubit(QuantumRegister(3, 'q'), 2)
+                }), input_qubit_mapping={Qubit(QuantumRegister(3, 'q'), 0): 0,
+                Qubit(QuantumRegister(3, 'q'), 1): 1, Qubit(QuantumRegister(3,
+                'q'), 2): 2}, final_layout=None, _input_qubit_count=3,
+                _output_qubit_list=[Qubit(QuantumRegister(3, 'q'), 0), Qubit
+                (QuantumRegister(3, 'q'), 1), Qubit(QuantumRegister(3, 'q'),
+                2)])
+
+
+
+
         """
         return self._layout
 
     @property
     def data(self) -> QuantumCircuitData:
         """The circuit data (instructions and context).
 
-        Returns:
-            QuantumCircuitData: a list-like object containing the :class:`.CircuitInstruction`\\ s
-            for each instruction.
+        Example:
+
+            .. plot::
+                :include-source:
+                :nofigs:
+
+                from qiskit import QuantumCircuit
+
+                qc = QuantumCircuit(2, 2)
+                qc.measure([0], [1])
+                print(qc.data)
+
+            .. code-block:: text
+
+                [CircuitInstruction(operation=Instruction(name='measure', num_qubits=1,
+                num_clbits=1, params=[]), qubits=(Qubit(QuantumRegister(2, 'q'), 0),),
+                clbits=(Clbit(ClassicalRegister(2, 'c'), 1),))]
+
+            Returns:
+                QuantumCircuitData: a list-like object containing the :class:`.CircuitInstruction`\\ s
+                for each instruction.
```

Comment:
Despite the previous docs, this shouldn't be in a "Returns" heading (also, this new "Returns" heading accidentally got nested under the "Example" heading), since this doesn't behave like a method.  We probably just want to replace the summary line with

> A list-like object containing the `` :class:`.CircuitInstruction` `` instances in the circuit.

The type `QuantumCircuitData` I think is generally private and undocumented (which is correct) - from a user's perspective, this should _look_ like a `list[CircuitInstruction]`.

Reply:
Good catch, thanks!